tensorflow/compiler/mlir/xla/tests/buffer-assignment.mlir

```diff
@@ -1,89 +1,145 @@
-// RUN: xla-opt -test-buffer-assignment -split-input-file %s | FileCheck %s -dump-input-on-failure
+// RUN: tf-opt -test-buffer-assignment-preparation -buffer-assignment -allow-unregistered-dialect -split-input-file %s | FileCheck %s -dump-input-on-failure
```

Comment:
Do you need to run both the ` -test-buffer-assignment-preparation ` and the ` -buffer-assignment` in this test? I don't see where you're checking any of the `-test-buffer-assignment-preparation` output anymore?

Reply:
Yes, as it is stated in the documentation of BA, the dialect-expert, who implements the dialect-lowering legalization, needs to use buffer assignment in two steps. 
The first step occurs during the legalization by using BA helper methods like computeAllocPosition or pattern converters (if needed) like FunctionAndBlockSignatureConverter or NonVoidToVoidReturnOpConverter. These helpers are tested with ```-test-buffer-assignment-preparation``` flag. 
In the second phase, dialect-expert needs to use ```-buffer-assignment``` flag to insert missing Dealloc operations, and move existing Alloc and Dealloc operations to their proper positions.
In ```test-buffer-assignment-preparation``` pass, test operations like

- buffer_assignment_test.unary
- buffer_assignment_test.unary_lowered
- buffer_assignment_test.copy

have been created to simulate legalization step in TF or MLIR independently from existing dialects. So, the positions of AllocOp and DeallocOp are tested by checking their surrounding lowered operations after legalization. Please note some tests, that are lowered, only use ```-buffer-assignment``` and some use both flags. So, what we can do is to add a pass manager to ```test-buffer-assignment-preparation``` pass and apply ```-buffer-assignment``` pass afterwards.

Reply:
I don't really follow here. I expect the two passes to be able to be shown independently on different IR with different CHECK. So what you can do is have two files: one that has the tests for `test-buffer-assignment-preparation` and another file with the tests for `test-buffer-assignment`

Reply:
All right. We are going to separate the test files in the next PR for BA like as it was at the beginning.
The reasons we introduced these fakes operations were first having operations that don't have region inside (BA doesn't support nested regions currently) and can be converted by a simple OpConversion. Secondly, making the transfer of BA to MLIR easier without adapting all test cases since Lhlo isn't available in MLIR. However, when BA supports nested regions, we are going to get rid of these fake operations by replacing them with Linalg ones as you mentioned.